File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> mp;

        for(int el : nums){
            mp[el]++;
        }

        priority_queue<pair<int,int>, vector<pair<int,int>>, greater<pair<int,int>>> pq;

        for(auto el : mp){
            pq.push({el.second, el.first});
            if(pq.size() > k) {
                pq.pop();
            }
        }

        vector<int> ans;
        while(!pq.empty()){
            ans.push_back(pq.top().second);
            pq.pop();
        }
        return ans;        
    }
};

File: Data Structures & Algorithms/sort-an-array/submission-1.java
class Solution {
    private void mergeSort(int[] nums, int start, int end){
        if(start>= end) return;

        int mid = start + (end-start)/2;

        mergeSort(nums, start, mid);
        mergeSort(nums, mid+1, end);

        int[] ans = new int[end-start+1];
        int k = ans.length -1;

        int i = mid;
        int j = end;
        while(i>=start && j>mid){
            if(nums[i]>nums[j]){
                ans[k] = nums[i];
                i--;
            } else {
                ans[k] = nums[j];
                j--;
            }
            k--;
        }
        while(i>=start){
            ans[k] = nums[i]; 
            i--;
            k--;
        }

        
        while(j>mid){
            ans[k] = nums[j]; 
            k--;
            j--;
        }
        k = 0;
        i = start;
        for(; i<=end; i++){
            nums[i] = ans[k++];
        }
    }
    public int[] sortArray(int[] nums) {
        mergeSort(nums, 0, nums.length-1);
        return nums;
    }
}